homeassistant/components/sia/binary_sensor.py

```diff
@@ -0,0 +1,163 @@
+"""Module for SIA Binary Sensors."""
+from __future__ import annotations
+
+from collections.abc import Iterable
+import logging
+from typing import Any
+
+from pysiaalarm import SIAEvent
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_MOISTURE,
+    DEVICE_CLASS_POWER,
+    DEVICE_CLASS_SMOKE,
+    BinarySensorEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_OFF, STATE_ON, STATE_UNAVAILABLE
+from homeassistant.core import HomeAssistant, State
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    CONF_ACCOUNT,
+    CONF_ACCOUNTS,
+    CONF_ZONES,
+    SIA_HUB_ZONE,
+    SIA_UNIQUE_ID_FORMAT_BINARY,
+)
+from .sia_entity_base import SIABaseEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+POWER_CODE_CONSEQUENCES: dict[str, bool] = {
+    "AT": False,
+    "AR": True,
+}
+
+SMOKE_CODE_CONSEQUENCES: dict[str, bool] = {
+    "GA": True,
+    "GH": False,
+    "FA": True,
+    "FH": False,
+    "KA": True,
+    "KH": False,
+}
+
+MOISTURE_CODE_CONSEQUENCES: dict[str, bool] = {
+    "WA": True,
+    "WH": False,
+}
+
+
+def generate_binary_sensors(entry) -> Iterable[SIABinarySensorBase]:
+    """Generate binary sensors.
+
+    For each Account there is one power sensor with zone == 0.
+    For each Zone in each Account there is one smoke and one moisture sensor.
+    """
+    for account in entry.data[CONF_ACCOUNTS]:
+        yield SIABinarySensorPower(entry, account)
+        zones = entry.options[CONF_ACCOUNTS][account[CONF_ACCOUNT]][CONF_ZONES]
+        for zone in range(1, zones + 1):
+            yield SIABinarySensorSmoke(entry, account, zone)
+            yield SIABinarySensorMoisture(entry, account, zone)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up SIA binary sensors from a config entry."""
+    async_add_entities(generate_binary_sensors(entry))
+
+
+class SIABinarySensorBase(BinarySensorEntity, SIABaseEntity):
```

Comment:
It's safer to put the integration base class first in the MRO since we pass things to the init method of the super class.